autogpt_platform/frontend/src/app/(platform)/build/components/NewBlockMenu/BlockMenuSidebar/useBlockMenuSidebar.ts

```diff
@@ -0,0 +1,20 @@
+import { useGetV2GetBuilderItemCounts } from "@/app/api/__generated__/endpoints/default/default";
+import { useBlockMenuContext } from "../block-menu-provider";
+
+export const useBlockMenuSidebar = () => {
+  const { defaultState, setDefaultState } = useBlockMenuContext();
+
+  const { data: blockCounts } = useGetV2GetBuilderItemCounts({
+    query : {
+        select : (x) =>{
+            return x.data
+        }
+    }
+  });
```

Comment:
What about the loading and failure handling? Maybe we should do? 💭 

- **On failure**
  - we should show a toast with the API error message?
- **On loading**
  - we can disable the interactions on the sidebar buttons?